controllers/managedcluster_controller.go

```diff
@@ -0,0 +1,122 @@
+	"fmt"
+	"log/slog"
+	"reflect"
+
+	"github.com/red-hat-storage/odf-multicluster-orchestrator/controllers/utils"
+	viewv1beta1 "github.com/stolostron/multicloud-operators-foundation/pkg/apis/view/v1beta1"
+	"k8s.io/apimachinery/pkg/types"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type ManagedClusterReconciler struct {
+	Client client.Client
+	Logger *slog.Logger
+}
+
+const odfConfigMapName = "odf-info"
+
+func (r *ManagedClusterReconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	logger := r.Logger.With("ManagedCluster", req.NamespacedName)
+	logger.Info("Reconciling ManagedCluster")
+
+	var managedCluster clusterv1.ManagedCluster
+	if err := r.Client.Get(ctx, req.NamespacedName, &managedCluster); err != nil {
+		if client.IgnoreNotFound(err) != nil {
+			logger.Error("Failed to get ManagedCluster", "error", err)
+		}
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if err := r.processManagedClusterViews(ctx, managedCluster.Name); err != nil {
+		logger.Error("Failed to ensure ManagedClusterView", "error", err)
+		return ctrl.Result{}, err
+	}
+
+	logger.Info("Successfully reconciled ManagedCluster", "name", managedCluster.Name)
+
+	return ctrl.Result{}, nil
+}
+
+func (r *ManagedClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	r.Logger.Info("Setting up ManagedClusterReconciler with manager")
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&clusterv1.ManagedCluster{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Watches(
+			&viewv1beta1.ManagedClusterView{},
+			handler.EnqueueRequestsFromMapFunc(r.managedClusterViewRequestMapper),
+			builder.WithPredicates(managedClusterViewStatusChangePredicate, predicate.GenerationChangedPredicate{}),
+		).
+		Complete(r)
+}
+
+func (r *ManagedClusterReconciler) managedClusterViewRequestMapper(ctx context.Context, obj client.Object) []reconcile.Request {
+	mcv, ok := obj.(*viewv1beta1.ManagedClusterView)
+
+	if !ok {
+		r.Logger.Error("Received object is not a ManagedClusterView object")
+		return []reconcile.Request{}
+	}
+	logger := r.Logger.With("ManagedClusterView", mcv.Name)
+	mcName, ok := mcv.Labels[utils.MCVLabelKey]
+
+	if !ok {
+		return []reconcile.Request{}
+	}
+
+	logger.Debug("ManagedClusterView of interest just got updated. Raising a reconcile request for ManagedCluster", "ManagedCluster", mcName)
+
+	return []reconcile.Request{
+		{
+			NamespacedName: types.NamespacedName{
+				Name: mcName,
+			},
+		},
+	}
+
+}
+func (r *ManagedClusterReconciler) processManagedClusterViews(ctx context.Context, clusterName string) error {
+	namespace := "openshift-storage"
```

Comment:
There was a ROSA epic as per which now ODF/OCS/Console can work in any namespace.
For MCO is it still always `openshift-storage` ??